sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryPipelineResponse.java

```diff
@@ -33,28 +33,32 @@ public String getBody() {
         return body;
     }
 
-    public Set<String> getErrors() {
+    public Set<ResponseError> getErrors() {
         return parseErrors(body); // parseErrors on demand
     }
 
+    public Set<String> getErrorMessages() {
+        return getErrors().stream().map(ResponseError::getMessage).collect(Collectors.toSet());
+    }
+
     public boolean isInvalidInstrumentationKey() {
-        Set<String> errors = parseErrors(body);
+        Set<String> errors = getErrorMessages();
         return errors != null && errors.contains(INVALID_INSTRUMENTATION_KEY);
     }
 
-    private static Set<String> parseErrors(String body) {
+    static Set<ResponseError> parseErrors(String body) {
         JsonNode jsonNode;
         try {
             jsonNode = new ObjectMapper().readTree(body);
         } catch (JsonProcessingException e) {
             // fallback to generic message
-            return singleton("Could not parse response");
+            return singleton(null);
```

Comment:
I think we're losing some logging that we used to get from this

maybe let `JsonProcessingException` bubble up from this method and handle it in `getErrorMessage` to return `singleton("Could not parse response")`

and in `getErrors` can ignore it and return `emptSet()`